nanoPU_sim.py

```diff
@@ -217,6 +203,16 @@ def start(self):
                                            pkt[NDP].pkt_offset),
                             pkt[NDP].payload)
                     self.assemble_queue.put(data)
+                    pull_offset_diff = 1
```

Comment:
Just want to check: the credit is supposed to increase with _every_ DATA pkt and not every _new_ (i.e. unduplicated) DATA pkt?

Reply:
I am aware that it is a bit odd, but yes this was how the original NDP simulation was implemented. The idea is the following:
- If a data packet is received, it means the network can handle a new data packet now, so the receiver pulls a new data packet. 
- After all, if an old packet is being received for the second time, this means the sender timed out in the first time. This occurs when the ACK sent by the receiver regarding the first packet was lost which means the congestion is in the reverse path, so pulling a new packet is not harmful.

Reply:
Ah ok, thanks for that.

We don't need to add this to this PR, but I think it would be helpful to change the `getRxMsgID` extern call to `getRxMsgInfo` which returns: `rx_msg_id`, `is_new_msg`,  `is_new_pkt`, and `ack_no`. Where `is_new_pkt` indicates if this pkt has been received before. This would be useful for implementing Homa to allow the ingress pipeline to compute remaining msg size.

Reply:
Agreed. Let's do that once we merge this PR. I can do that.